### PR TITLE
Ignore table without primary key 

### PR DESCRIPTION
**Pull request by perfectacle**
When you sync table without primary key,  
some attribute(skip_non_pk) will help you.

### CODE REVIEW

**Comment by siddontang** [comment]
@perfectacle 

Any benefit to support this feature? If we don't have a PK, how to get the data directly from ES?

**Comment by perfectacle** [comment]
If you use this attribute, go-mysql-elasticsearch doesn't sync table without PK.  
so you don't need to find data in ignored table.  
This is a good match for * wildcard,  
if you use * wildcard without this property, you would see below error,  
... table must have a PK for a column

so you should add all table manually except table without PK,
but if you use this feature, it works automatically and logging what table doesn't has PK

**Comment by siddontang** [comment]
So why do we allow syncing a table without a PK? 

I guess here you just want to let the syncer still work well, not close when meeting this error.  Am I right?

PTAL @EagleChen

**Comment by perfectacle** [comment]
@siddontang nono, this feature ignore table without a PK,
ex) tables = ["tbl1", "tbl2", "tbl3"]
and tbl2 doesn't has pk,
then sync tbl1, tbl3
it works same to below
tables = ["tbl1", "tbl3"]

**Comment by siddontang** [comment]
@perfectacle 

Your example shows the same thing - don't sync the table which has no PK. 

So here I suggest using a configuration like SkipNoPKTable to go on if we meet this case.

**Comment by perfectacle** [comment]
you mean 
Ignore table without primary key(x)
Don't sync the table which has no PK(O)

right...?
so I need to change docs and property name... right...?

File: README.md [hunk 1]
```diff
@@ -150,6 +150,18 @@ filter = ["id", "name"]
...
+## Ignore table without primary key
+When you sync table without primary key, you can see below error message.
+```
+.../go-mysql-elasticsearch/river/river.go:244: schema.table must have a PK for a column
+```
+If you want to ignore table without primary key, you can ignore by adding this attribute to the *.toml.  
+
+```
+# Ignore table without primary key
+skip_non_pk = true
```
Comment (siddontang) [inline README.md]: skip_no_pk_table

File: river/river.go [hunk 2]
```diff
@@ -232,16 +232,23 @@ func (r *River) prepareRule() error {
...
-	for _, rule := range r.rules {
+	rules := make(map[string]*Rule)
+	for key, rule := range r.rules {
 		if rule.TableInfo, err = r.canal.GetTable(rule.Schema, rule.Table); err != nil {
 			return errors.Trace(err)
 		}
 
 		if len(rule.TableInfo.PKColumns) == 0 {
-			return errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table)
+			if r.c.SkipNonPk == false {
```
Comment (siddontang) [inline river/river.go]:
```
if !r.c.SkipNoPKTable  {
    return xxx
}

log.Errorf()
```

File: README.md [hunk 3]
```diff
@@ -150,6 +150,18 @@ filter = ["id", "name"]
 
 In the above example, we will only sync MySQL table tfiler's columns `id` and `name` to Elasticsearch. 
 
+## Ignore table without primary key
```
Comment (siddontang) [inline README.md]: without a primary key

File: README.md [hunk 4]
```diff
@@ -150,6 +150,18 @@ filter = ["id", "name"]
...
+When you sync table without primary key, you can see below error message.
+```
+.../go-mysql-elasticsearch/river/river.go:244: schema.table must have a PK for a column
```
Comment (siddontang) [inline README.md]: remove `.../go-mysql-elasticsearch/river/river.go:244: `

File: README.md [hunk 5]
```diff
@@ -150,6 +150,18 @@ filter = ["id", "name"]
...
+```
+If you want to ignore table without primary key, you can ignore by adding this attribute to the *.toml.
```
Comment (siddontang) [inline README.md]: You can ignore these tables in the configuration like:

File: river/river.go [hunk 6]
```diff
@@ -232,16 +232,23 @@ func (r *River) prepareRule() error {
...
+				return errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table)
+			} else {
+				fmt.Printf("ignored table without primary key: %s\n", rule.TableInfo.Name)
```
Comment (siddontang) [inline river/river.go]: use log.Errorf

Comment (perfectacle) [reply river/river.go]: errors.Errorf("%s.%s must have a PK for a column", rule.Schema, rule.Table) to log.Errorf...?
or fmt.Printf("ignored table without primary key: %s\n", rule.TableInfo.Name) to log.Errorf...?

File: river/river.go (hunk 2 above)
Comment (perfectacle) [reply river/river.go]: I can't understand...  
`if !r.c.SkipNoPKTable {}` alright...
but return xxx...??
and where to write log.Errorf...??

plz give me some detail information!!

File: river/river.go (hunk 6 above)
Comment (siddontang) [reply river/river.go]: fmt.Printf -> log.Errorf

**Comment by perfectacle** [comment]
hey @siddontang I edited it!!
please check it!

**Comment by siddontang** [comment]
LGTM

**Comment by EagleChen** [comment]
LGTM. Sorry for the late response.

**Pull request merged**